### PR TITLE
Static CRT and detect entrypoint by link.exe

### LINKED ISSUE #446: Add an option to include VC++ runtime for --native
On windows you're required to install the VC++ 2014 runtime to get the native executable working, there should be an option/parameter to include the runtime statically.

### PR DESCRIPTION
**Pull request by schellap**
Fixes #446 

@gkhanna79, is an explicit option to include CRT for Windows only really necessary?

### CODE REVIEW

**Comment by gkhanna79** [comment]
Windows Release build is failing with the following:

PortableRuntime.lib(portable.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in e2etestroot.obj

Same with Debug build:

PortableRuntime.lib(portable.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MT_StaticRelease' in e2etestroot.obj

File: src/Microsoft.DotNet.Tools.Compiler.Native/IntermediateCompilation/Windows/WindowsCppCompileStep.cs [hunk 1]
```diff
@@ -22,8 +22,8 @@ public class WindowsCppCompileStep : IPlatformNativeStep
         
         private static readonly Dictionary<BuildConfiguration, string> ConfigurationCompilerOptionsMap = new Dictionary<BuildConfiguration, string>
         {
-            { BuildConfiguration.debug, "/ZI /nologo /W3 /WX- /sdl /Od /D CPPCODEGEN /D WIN32 /D _CONSOLE /D _LIB /D _UNICODE /D UNICODE /Gm /EHsc /RTC1 /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Gd /TP /wd4477 /errorReport:prompt" },
-            { BuildConfiguration.release, "/Zi /nologo /W3 /WX- /sdl /O2 /Oi /GL /D CPPCODEGEN /D WIN32 /D NDEBUG /D _CONSOLE /D _LIB /D _UNICODE /D UNICODE /Gm- /EHsc /MD /GS /Gy /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Gd /TP /wd4477 /errorReport:prompt" }
+            { BuildConfiguration.debug, "/ZI /nologo /W3 /WX- /sdl /Od /D CPPCODEGEN /D WIN32 /D _CONSOLE /D _LIB /D _UNICODE /D UNICODE /Gm /EHsc /RTC1 /MT /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Gd /TP /wd4477 /errorReport:prompt" },
```
Comment (gkhanna79) [inline src/Microsoft.DotNet.Tools.Compiler.Native/IntermediateCompilation/Windows/WindowsCppCompileStep.cs]: For Debug, I think you need to specific "/MTd"

**Comment by gkhanna79** [comment]
> is an explicit option to include CRT for Windows only really necessary?

Which option are you referring to?

I also think you need to remove the following section, from WindowsLinkSteps.cs, that links msvcrt.lib now that you plan to link against static CRT:

private static readonly Dictionary<BuildConfiguration, string[]> ConfigurationLinkLibMap = new Dictionary<BuildConfiguration, string[]>()

**Comment by schellap** [comment]
@gkhanna79, we need to compile our libraries with MT and MTd as well for these errors to go away.

**Comment by gkhanna79** [comment]
@schellap Lets chat about this offline

**Comment by schellap** [comment]
@dotnet-bot test this please

**Comment by schellap** [comment]
@gkhanna79 Tracked down the problem. In addition to dotnet native, MCG also has declared a dependency on ILC and that was stomping on the ILC version brought in, still using the old libs. Hence the CI failed.

**Comment by gkhanna79** [comment]
Thanks. Can you please file an issue against MCG to declare dependency on the same (latest published) version of ILC as we do?

**Comment by schellap** [comment]
Clarified with Tijoy that it is not a needed dependency.

**Comment by schellap** [comment]
@dotnet-bot test this please

**Comment by schellap** [comment]
Closing PR as the dotnet-bot won't test this PR.

**Pull request closed without merging**